### PR TITLE
refactor(governance)!: Aligned package with updates to CIP-36

### PR DESCRIPTION
**Pull request by Ryun1**
# Context

[CIP-36 (Catalyst/Voltaire Registration Transaction Metadata Format (Updated))](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0036) has seen some a couple _recent_ changes:
- [#373](https://github.com/cardano-foundation/CIPs/pull/373) adjusted the metadata standard, replacing the `reward_address` field with `payment_address`. Where `payment_address` contains a payment address rather than stake/rewards address. To accompany this new test vectors were supplied.
- [#427](https://github.com/cardano-foundation/CIPs/pull/427) focussed on adjusting semantics to reduce ambiguity for tool makers and potential future protocol governance. Adding clarity around naming of vote keys (see [here](https://github.com/cardano-foundation/CIPs/blob/master/CIP-0036/README.md?plain=1#L60)).

# Proposed Solution
- Alter the governance package to reflect the changes introduced in [#373](https://github.com/cardano-foundation/CIPs/pull/373):
  - Replaced rewards address with payment address.
  - Added new test vectors, to match those provided in the CIP.
- Alter the governance package to reflect the naming conventions introduced in [#427](https://github.com/cardano-foundation/CIPs/pull/427):
  - Renaming elements that use naming such as `votingKey` to use `CIP36VoteKey`.
  - Altering comments that use terms such as "voting key" to use "vote key".

# Important Changes Introduced
- **cip36.ts**: 
  - Renamed `interface VotingKeyDerivationPath` to be `interface CIP36VoteKeyDerivationPath`.
  - renamed `interface GovernanceKeyDelegation` to be `interface VoteKeyDelegation`.
  - In `interface GovernanceKeyDelegation` I renamed `votingKey` to be `cip36VoteKey`.
  - In `interface BuildVotingRegistrationProps` I replaced `rewardAccount` with `paymentAddress` and changed the type to be a `Cardano.Address` rather than `Cardano.RewardAccount`.
  - Changed comments to reflect.
- **cip36.test.ts**: 
  - Matched test vectors used in [CIP-36](https://github.com/cardano-foundation/CIPs/tree/master/[CIP-0036](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0036)).
  - Matched field changes from **cip36.ts**.
- **cip36KeyAgents.test.ts**: 
  - Matched field changes from **cip36.ts**.
  - Renamed instances of `votingKey` to `CIP36VoteKey`.
   - Changed comments to reflect.

### CODE REVIEW

File: packages/governance/src/cip36.ts [hunk 1]
```diff
@@ -17,18 +17,18 @@ export enum MetadataLabel {
...
 }
 
 /**
- * Voting power delegation to a specified voting key
+ * Voting power delegation to a specified vote key
  */
-export interface GovernanceKeyDelegation {
-  votingKey: Cardano.Ed25519PublicKey;
+export interface VoteKeyDelegation {
+  CIP36VoteKey: Cardano.Ed25519PublicKey;
```
Comment (mkazlauskas) [inline packages/governance/src/cip36.ts:31]: Let's stick to this naming convention
```suggestion
  cip36VoteKey: Cardano.Ed25519PublicKey;
```

**Review by mkazlauskas — CHANGES_REQUESTED** [review]
Thanks @Ryun1! Please update commit message to `refactor(governance)!: aligned with updates to cip-36` to indicate that it has breaking changes

File: packages/governance/src/cip36.ts (hunk 1 above)
Comment (Ryun1) [reply packages/governance/src/cip36.ts:31]: Good idea, I will make this change and amend the commit message

**Review by rhyslbw — CHANGES_REQUESTED** [review]
Looks good @Ryun1. There's a conflict after https://github.com/input-output-hk/cardano-js-sdk/pull/589 was merged earlier, so you'll need to rebase this branch.

**Review by rhyslbw — APPROVED** [review]

**Review by mkazlauskas — APPROVED** [review]

**Pull request merged**